single-kernel/lin_reg_coeff.cpp

```diff
@@ -0,0 +1,194 @@
+                  local_mem[local_mem_index] = local_mem[local_mem_index] + local_mem[local_mem_index + stride];
+              }
+
+              item.barrier(s::access::fence_space::local_space);
+            }
+
+            // Only one work-item per work group writes to global memory 
+            if (lid == 0) {
+              global_mem[item.get_group_linear_id()] = local_mem[0];
+            }
+          });
+      });
+    array_size = n_wgroups;
+  }
+  auto reduced_value = input_buf.template get_access<s::access::mode::read>();
+  return(reduced_value[0]);
+}
+
+  void run() {    
+    s::buffer<T, 1> input1_buf(input1.data(), s::range<1>(args.problem_size));
+    s::buffer<T, 1> input2_buf(input2.data(), s::range<1>(args.problem_size));
+    s::buffer<T, 1> output_buf(output.data(), s::range<1>(args.problem_size));
+
+    vec_product(input1_buf, input2_buf, output_buf);
+
+    T ss_xy = reduce(output_buf);
+
+    vec_product(input1_buf, input1_buf, output_buf);
+
+    T ss_xx = reduce(output_buf);
+
+    T mean_x = reduce(input1_buf)/args.problem_size;
+    T mean_y = reduce(input2_buf)/args.problem_size;
+
+    ss_xy = ss_xy - mean_x*mean_y;
+    ss_xx = ss_xx - mean_x*mean_x;
+
+    coeff_b1 = ss_xy/ss_xx;
+    coeff_b0 = mean_y - coeff_b1*mean_x;
+
+    //std::cout << "ss_xy = " << ss_xy << "ss_xx = " << ss_xx << std::endl;
+    //std::cout << "Mean_x = " << mean_x << "Mean_y = " << mean_y << std::endl;
+    //std::cout << "Coeff_b1 = " << coeff_b1 << ", " << "Coeff_b0 = " << coeff_b0 << std::endl;
+  }
+
+  bool verify(VerificationSetting &ver) { 
+     bool pass = true;
+    
+    T sum_of_vec1 = 0;
+    T sum_of_vec2 = 0;
+    for (size_t i = 0; i < args.problem_size; i++) {
+      sum_of_vec1 += input1ver[i];
+      sum_of_vec2 += input2ver[i];
+    }
+
+    T mean_x = sum_of_vec1/args.problem_size;
+    T mean_y = sum_of_vec2/args.problem_size;
+
+    T ss_xy = 0;
+    T ss_xx = 0;
+    for (size_t i = 0; i < args.problem_size; i++) {
+      ss_xy += input1ver[i]*input2ver[i];
+      ss_xx += input1ver[i]*input1ver[i];
+    }
+
+    ss_xy = ss_xy - mean_x*mean_y;
+    ss_xx = ss_xx - mean_x*mean_x;
+
+    T expected_coeff_b1 = ss_xy/ss_xx;
+    T expected_coeff_b0 = mean_y - expected_coeff_b1*mean_x;
+
+    std::cout << "Coeff_b1 = " << coeff_b1 << ", " << "Coeff_b0 = " << coeff_b0 << std::endl;
+    std::cout << "Expected Coeff_b1 = " << expected_coeff_b1 << ", " << "Expected Coeff_b0 = " << expected_coeff_b0 << std::endl;
+
+    const T tolerance = 0.00001;
+    if ((fabs(expected_coeff_b0 - coeff_b0) > tolerance) || (fabs(expected_coeff_b1 - coeff_b1) > tolerance))
+      pass = false;
+
+    return pass;
+  }
+  
+  static std::string getBenchmarkName() {
+    std::stringstream name;
+    name << "LinearRegressionCoeff_";
+    name << ReadableTypename<T>::name;
+    return name.str();     
+  }
+};
+
+int main(int argc, char** argv)
+{
+  BenchmarkApp app(argc, argv);
+  app.run<LinearRegressionCoeffBench<float>>();
```

Comment:
This should be surrounded with `if(app.shouldRunNDRangeKernels())` since this benchmark uses hierarchical parallel for. This will allow us to disable the benchmark when the `--no-ndrange-kernels` command line option is used.

Reply:
hi Aksel, thanks for the quick comment. The current version does not use the hierarchical parallel for, but I will do that.